### PR TITLE
Added support for dismissed_review IssueEvent

### LINKED ISSUE #827: Support dismissed_review IssueEvent
GitHub Developer API announcement:
https://developer.github.com/changes/2018-01-05-dismiss-event-breaking-change/

This announcement actually confuses me a bit...
From the announcement, the `dismissed_review` IssueEvent is changing, but I don't see any history of it, so I don't know when it was introduced. Nevertheless, we would like to support it, so it needs to be added to the `IssueEvent` struct in `github/issues_events.go`.

I think it would suffice to leave its type as a `string` field, similar to other states returned by this package (as opposed to strictly enumerating the values), unless there are objections.

This would be a great first issue for a new contributor to this repo or new Go developer to work on.
All help is greatly appreciated. Thank you!

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
fixes #827

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@gmlewis Please review 😄

File: github/issues_events.go [hunk 1]
```diff
@@ -69,14 +69,25 @@ type IssueEvent struct {
...
+	CommitID        *string          `json:"commit_id,omitempty"`
+	Milestone       *Milestone       `json:"milestone,omitempty"`
+	Label           *Label           `json:"label,omitempty"`
+	Rename          *Rename          `json:"rename,omitempty"`
+	LockReason      *string          `json:"lock_reason,omitempty"`
+	ProjectCard     *ProjectCard     `json:"project_card,omitempty"`
+	DismissedReview *DismissedReview `json:"dismissed_review,omitempty"`
+}
+
+// DismissedReview represents details for 'dismissed_review' events
```
Comment (gmlewis) [inline github/issues_events.go]: Please terminate comments with periods... here and below.

File: github/issues_events.go [hunk 2]
```diff
@@ -69,14 +69,25 @@ type IssueEvent struct {
...
+type DismissedReview struct {
+	// State represents the state of the dismissed review and
```
Comment (gmlewis) [inline github/issues_events.go]: How about:

```go
// State represents the state of the dismissed review.
// Possible values are: "commented", "approved", and "changes_requested".
```

File: github/issues_events.go [hunk 3]
```diff
@@ -69,14 +69,25 @@ type IssueEvent struct {
...
+	// possible state values are: commented, approved and changes_requested
+	State            *string `json:"state,omitempty"`
+	ReviewID         *int64  `json:"review_id,omitempty"`
+	DismissalMessage *string `json:"dismissal_message,omitempty"`
+	DismissalCommit  *string `json:"dismissal_commit_id,omitempty"`
```
Comment (gmlewis) [inline github/issues_events.go]: s/DismissalCommit/DismissalCommitID/

File: github/issues_events.go [hunk 4]
```diff
@@ -69,14 +69,25 @@ type IssueEvent struct {
...
-	ProjectCard *ProjectCard `json:"project_card,omitempty"`
+	Assignee        *User            `json:"assignee,omitempty"`
+	Assigner        *User            `json:"assigner,omitempty"`
+	CommitID        *string          `json:"commit_id,omitempty"`
+	Milestone       *Milestone       `json:"milestone,omitempty"`
+	Label           *Label           `json:"label,omitempty"`
+	Rename          *Rename          `json:"rename,omitempty"`
+	LockReason      *string          `json:"lock_reason,omitempty"`
+	ProjectCard     *ProjectCard     `json:"project_card,omitempty"`
+	DismissedReview *DismissedReview `json:"dismissed_review,omitempty"`
```
Comment (gmlewis) [inline github/issues_events.go]: Please add documentation for this new event type in the comments above after line 65. Something like:

```go
//    review_dismissed
//      The review was dismissed and `DismissedReview` will be populated below.
```

File: github/issues_events.go (hunk 2 above)
Comment (vaibhavsingh97) [reply github/issues_events.go]: I had updated the PR 👍

File: github/issues_events.go (hunk 1 above)
Comment (vaibhavsingh97) [reply github/issues_events.go]: Sure, will keep in mind in future 😄

File: github/issues_events.go (hunk 3 above)
Comment (vaibhavsingh97) [reply github/issues_events.go]: Updated the PR

File: github/issues_events.go (hunk 4 above)
Comment (vaibhavsingh97) [reply github/issues_events.go]: Updated the PR

**Comment by vaibhavsingh97** [comment]
@gmlewis I had updated the PR, please review 😄

File: github/issues_events.go [hunk 5]
```diff
@@ -63,6 +63,8 @@ type IssueEvent struct {
 	//     head_ref_deleted, head_ref_restored
 	//       The pull request’s branch was deleted or restored.
 	//
+	//    review_dismissed
+	//      The review was dismissed and `DismissedReview` will be populated below.
```
Comment (gmlewis) [inline github/issues_events.go]: Just to be consistent, let's please add back in the one more blank line comment here, like line 65 above.

**Comment by vaibhavsingh97** [comment]
@gmlewis updated the PR 😄

**Comment by gmlewis** [comment]
Thank you, @gauntface!
Merging.

**Pull request merged**